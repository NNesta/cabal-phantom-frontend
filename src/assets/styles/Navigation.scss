.logo {
      cursor: pointer;
      color: #fff;
      font-weight: bold;
      font-size: 2.3rem;
      letter-spacing: 2px;
}

#nav_black {
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: fixed;
  width: 100%;
  z-index: 100;
  padding: 1.3rem 2rem;
  box-shadow: 0 2px 5px 0 rgba(0,0,0,0.4);
}

nav ul {
      display: flex;
      align-items: center;
      list-style: none;
      overflow: hidden;
      color: #fff;
      padding: 0;
      margin: 0;
      transition: 1s;
}

nav ul li{
      margin: 0 1.5rem;
}

.black {
      background:#040F23;
}

nav ul li a {
      text-decoration: none;
      color: #fff;
      font-size: 1.6rem;
      font-weight: normal;
      cursor: pointer;

      &:hover{
            color: #ccc;
            animation: fill 1s forwards;
      }
}

.menu-icon {
      box-sizing: border-box;
      cursor: pointer;
      color: #fff;
      display: none;
}

.headers {
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: fixed;
  width: 100%;
  z-index: 100;
  padding: 1.2rem 0;
  border-bottom: 1px solid #ccc;
  box-shadow: 0 4px 2px -2px #ccc;
}
.headers__menu{
  padding-right: 2rem;
}
.headers__logo {
  font-size: 3rem;
  font-weight: bold;
  letter-spacing: 0.11rem;
  color: #fff;
  padding-left: 2rem;
  font-family: Roboto;
}
.headers__menu__item {
  margin: 0 1rem;
  color: #fff;
  text-decoration: none;
  letter-spacing: 0.11rem;
  font-size: 2.4rem;
  font-family: Roboto;
}
.headers__menu__select, li>select{
  padding: 0.8rem;
  border-radius: 0.4rem;
  outline: none;
  background: #fff;
  border: 1px solid #fff;
  font-size: 1.4rem;
}

.login_menu{
  padding: 0.8rem 2rem;
  border-radius: 0.4rem;
  border: 1px solid #fff;
  font-size: 1.4rem;

  &:hover{
   background: #040F23;
   text-decoration: none;
  }
}

@media(max-width: 786px) {
      .login_menu{
            border: none;
            box-shadow: 0 2px 5px 0 rgba(0,0,0,0.4);

      }

      .responsive-nav{
            width: 100%;
            display: flex;
            justify-content: space-between;
            align-items: center;
            padding: 1.3rem 2rem;
            box-shadow: 0 2px 5px 0 rgba(0,0,0,0.4);
      }

      #nav_black {
            display: flex;
            flex-direction: column;
            justify-content: center;
            align-items: center;
            box-shadow: none;
            padding: 0;
      }

      nav ul {
            max-height: 0px;
            background-color: #040F23;
            display: flex;
            flex-direction: column;
      }

      .menu{
            width: 100%;
      }

      .showing {
            max-height: 34em;
      }

      nav ul li {
            box-sizing: border-box;
            width: 100%;
            padding: .8rem 0;
            box-shadow: 0 2px 5px 0 rgba(0,0,0,0.4);
            text-align: center;
            
      }

      .menu-icon {
            display: block;
      }

}
