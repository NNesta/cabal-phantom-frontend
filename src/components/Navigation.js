import React, { Component } from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';

export default class Navigation extends Component {
    state = {
        backgroundColor: 'transparent'
    }

    listenScrollEvent = e => {
        if (window.scrollY > 20) {
            this.setState({ backgroundColor: '#040F23' })
        } else {
            this.setState({ backgroundColor: 'transparent' })
        }
    }

    componentDidMount() {
        window.addEventListener('scroll', this.listenScrollEvent)
    }
    render() {
        return (
            <div className="headers" style={{ backgroundColor: this.state.backgroundColor }}>
                <Link to='/'><div className="headers__logo">Phantom</div></Link> 
                <div className="headers__menu">
                    <select name="Languages" className="headers__menu__select">
                        <option value="English">English</option>
                        <option value="French" >French</option>
                        <option value="Kinyarwanda" >Kinyarwanda</option>
                    </select>
                </div>
            </div>
        )
    }
}
