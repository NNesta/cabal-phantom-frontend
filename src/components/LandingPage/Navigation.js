import React, { Component } from 'react'
import { BrowserRouter as Router, Link as Links } from "react-router-dom";
import { animateScroll as scroll, Link } from 'react-scroll'
import './../../assets/styles/Navigation.scss'

export default class Navigation extends Component {
    constructor(props) {
        super(props);
        this.state = {
            language: 'english',
            isActive: false,
            className: "menu-icon",
            hide: "hide"
        };
    }
    componentDidMount() {

        const scroller = () => {

            var nav1 = document.getElementById("nav_black");

            if (window.scrollY > 0) {
                nav1.classList.add("black");
            }
            else {
                nav1.classList.remove("black");
        }
        }
        window.addEventListener('scroll', scroller)
    }
    hiden = () => {
        if (this.state.hide === "hide") {
            this.setState({ hide: "showing" });
        }
        else {
            this.setState({ hide: "hide" });
        }
    }
    render() {
        return (
            <div id="wrapper" className="wrapper">
                <nav id="nav_black" className={this.state.isActive ? "black" : null}>
                    <div className="responsive-nav">
                        <div>
                            <Links to="/" >
                                <div className="logo" onClick={() => scroll.scrollToTop()}>
                                    Phantom
                                </div>
                            </Links>
                        </div>

                        <div className={this.state.className} id="icon" onClick={this.hiden}>
                            <i className="fa fa-bars fa-4x"></i>
                        </div>
                        
                    </div>
                    <div className="menu">
                        <ul id="nav_ul" className={this.state.hide}>
                            <li>
                                <Link
                                    className="header__menu__item"
                                    to="About"
                                    spy={true}
                                    smooth={true}
                                    offset={-50}
                                    duration={1000}>About Us
                                    </Link>
                            </li>
                            <li>
                                <Link
                                    className="header__menu__item"
                                    spy={true}
                                    smooth={true}
                                    offset={-50}
                                    duration={1000}
                                    to="feedback">Contact Us
                                    </Link>
                            </li>
                            <li className="login_menu">
                                <Links
                                    className="header__menu__item"
                                    to="/login"
                                    duration={1000}
                                >Login
                                </Links>
                            </li>
                            <li>
                                <select name="Languages" className="header__menu__select" value={this.setState.language}>
                                    <option value>English</option>
                                    <option value>French</option>
                                    <option value>Kinyarwanda</option>
                                </select>
                            </li>
                        </ul>
                    </div>
                </nav>
            </div>
        )
    }
}

