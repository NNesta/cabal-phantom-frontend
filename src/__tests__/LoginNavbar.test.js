import Navigation from '../components/Navigation';
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import { BrowserRouter as Router } from 'react-router-dom';


const NavigationComponent = () => {
    return render(
        <Provider store={store}>
            <Router>
                <Navigation />
            </Router>
        </Provider>
    );
};

describe('Navigation component', () => {
    afterEach(cleanup)

    it('Should render navigation snapshot component properly', () => {
        const { asFragment } = NavigationComponent();
        expect(asFragment(<Navigation />)).toMatchSnapshot();
    });
})
