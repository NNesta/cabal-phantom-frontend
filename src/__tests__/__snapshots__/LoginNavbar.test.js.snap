// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component Should render navigation snapshot component properly 1`] = `
<DocumentFragment>
  <div
    class="headers"
    style="background-color: transparent;"
  >
    <a
      href="/"
    >
      <div
        class="headers__logo"
      >
        Phantom
      </div>
    </a>
    <div
      class="headers__menu"
    >
      <select
        class="headers__menu__select"
        name="Languages"
      >
        <option
          value="English"
        >
          English
        </option>
        <option
          value="French"
        >
          French
        </option>
        <option
          value="Kinyarwanda"
        >
          Kinyarwanda
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
